<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DialogPreferences</class>
 <widget class="QDialog" name="DialogPreferences">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>485</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Preferences</string>
  </property>
  <property name="sizeGripEnabled">
   <bool>true</bool>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QGridLayout" name="gridLayout_6">
   <item row="0" column="0">
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Deviser Settings</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>The next three entries are needed, to generate the package / latex sources: </string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QWidget" name="widget1" native="true">
         <layout class="QGridLayout" name="gridLayout_3">
          <property name="leftMargin">
           <number>12</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="lblPython">
            <property name="text">
             <string>Python Interpreter:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <widget class="QLineEdit" name="txtPython"/>
            </item>
            <item>
             <widget class="QToolButton" name="cmdBrowsePython">
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="lblDeviserRepo">
            <property name="text">
             <string>Deviser Repository:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <widget class="QLineEdit" name="txtDeviserRepo"/>
            </item>
            <item>
             <widget class="QToolButton" name="cmdBrowseRepo">
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="lblOutputDir">
            <property name="minimumSize">
             <size>
              <width>160</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>Default output dir:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout_8">
            <item>
             <widget class="QLineEdit" name="txtOutputDir">
              <property name="toolTip">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span&gt;The directory where any output will be placed.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="cmdBrowseOutputDir">
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>The next two entries are needed, in order to compile the latex sources:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QWidget" name="widget2" native="true">
         <layout class="QGridLayout" name="gridLayout_2">
          <property name="leftMargin">
           <number>12</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="lblSbmlSpecDir">
            <property name="minimumSize">
             <size>
              <width>160</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>sbmlpkgspec dir:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout_9">
            <item>
             <widget class="QLineEdit" name="txtSbmlPkgSpec"/>
            </item>
            <item>
             <widget class="QToolButton" name="cmdBrowseSpec">
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="lblMikTexDir">
            <property name="text">
             <string>MikTeX bin dir:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout_11">
            <item>
             <widget class="QLineEdit" name="txtMikTexBinDir"/>
            </item>
            <item>
             <widget class="QToolButton" name="cmdBrowseMikTex">
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The next entries are needed, to compile the package with libSBML:&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QWidget" name="widget3" native="true">
         <layout class="QGridLayout" name="gridLayout">
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="lblCompiler">
            <property name="text">
             <string>Compiler:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout_13">
            <item>
             <widget class="QLineEdit" name="txtCompiler"/>
            </item>
            <item>
             <widget class="QToolButton" name="cmdBrowseCompiler">
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="lblSBMLSourceDir">
            <property name="text">
             <string>libSBML Source dir:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout_12">
            <item>
             <widget class="QLineEdit" name="txtLibSBMLSource"/>
            </item>
            <item>
             <widget class="QToolButton" name="cmdBrowseLibSBMLSource">
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="lblDependencies">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>160</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>Dependencies Source dir:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout_15">
            <item>
             <widget class="QLineEdit" name="txtDependencies"/>
            </item>
            <item>
             <widget class="QToolButton" name="cmdBrowseDeps">
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="lblCmakeExecutable">
            <property name="text">
             <string>CMake Executable:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout_14">
            <item>
             <widget class="QLineEdit" name="txtCMake"/>
            </item>
            <item>
             <widget class="QToolButton" name="cmdBrowseCMake">
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="lblSwigExecutable">
            <property name="text">
             <string>SWIG Executable:</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout_16">
            <item>
             <widget class="QLineEdit" name="txtSwig"/>
            </item>
            <item>
             <widget class="QToolButton" name="cmdBrowseSwig">
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="lblPythonIncludes">
            <property name="text">
             <string>Python include dir:</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout_17">
            <item>
             <widget class="QLineEdit" name="txtPythonIncludes"/>
            </item>
            <item>
             <widget class="QToolButton" name="cmdBrowsePythonIncludes">
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="lblPythonLib">
            <property name="text">
             <string>Python lib:</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout_18">
            <item>
             <widget class="QLineEdit" name="txtPythonLib"/>
            </item>
            <item>
             <widget class="QToolButton" name="cmdBrowsePythonLib">
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item row="6" column="0">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>5</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>User Defined Types</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="0" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>WARNING: Types added here will not be supported by Deviser. TeX generation will work and use the name of the type as given. C++ code generation will produce invalid code by using FIXME_[type] as the data type; where [type] will be the name of the type as given. Generated code will need to be manually adjusted. The Deviser code generation can be easily adapted to accept a user defined types and instructions are given in Appendix C of the User Manual.</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignJustify|Qt::AlignVCenter</set>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QGroupBox" name="grpAttributes">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string/>
         </property>
         <property name="title">
          <string> User Defined types: </string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <property name="sizeConstraint">
           <enum>QLayout::SetMaximumSize</enum>
          </property>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QToolButton" name="cmdAddAttr">
              <property name="text">
               <string>&amp;Add</string>
              </property>
              <property name="icon">
               <iconset resource="../deviser-resources.qrc">
                <normaloff>:/images/edit_add.png</normaloff>:/images/edit_add.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="cmdDelAttr">
              <property name="text">
               <string>&amp;Del</string>
              </property>
              <property name="icon">
               <iconset resource="../deviser-resources.qrc">
                <normaloff>:/images/list_remove.png</normaloff>:/images/list_remove.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer_2">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QTableView" name="tblUserDefinedTypes">
            <property name="alternatingRowColors">
             <bool>true</bool>
            </property>
            <property name="sortingEnabled">
             <bool>true</bool>
            </property>
            <attribute name="horizontalHeaderStretchLastSection">
             <bool>true</bool>
            </attribute>
            <attribute name="verticalHeaderVisible">
             <bool>false</bool>
            </attribute>
            <attribute name="verticalHeaderCascadingSectionResizes">
             <bool>true</bool>
            </attribute>
            <attribute name="verticalHeaderDefaultSectionSize">
             <number>23</number>
            </attribute>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../deviser-resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>DialogPreferences</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>DialogPreferences</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdBrowseCMake</sender>
   <signal>clicked()</signal>
   <receiver>DialogPreferences</receiver>
   <slot>browseCMakeExecutable()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>608</x>
     <y>272</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdBrowseCompiler</sender>
   <signal>clicked()</signal>
   <receiver>DialogPreferences</receiver>
   <slot>selectCompiler()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>608</x>
     <y>244</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdBrowseDeps</sender>
   <signal>clicked()</signal>
   <receiver>DialogPreferences</receiver>
   <slot>browseDependencies()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>608</x>
     <y>300</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdBrowseLibSBMLSource</sender>
   <signal>clicked()</signal>
   <receiver>DialogPreferences</receiver>
   <slot>browseLibSBML()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>608</x>
     <y>216</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdBrowseMikTex</sender>
   <signal>clicked()</signal>
   <receiver>DialogPreferences</receiver>
   <slot>browseMikTex()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>605</x>
     <y>169</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdBrowseOutputDir</sender>
   <signal>clicked()</signal>
   <receiver>DialogPreferences</receiver>
   <slot>browseOutputDir()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>605</x>
     <y>94</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdBrowsePython</sender>
   <signal>clicked()</signal>
   <receiver>DialogPreferences</receiver>
   <slot>browsePython()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>605</x>
     <y>38</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdBrowsePythonIncludes</sender>
   <signal>clicked()</signal>
   <receiver>DialogPreferences</receiver>
   <slot>browsePythonInclude()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>608</x>
     <y>356</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdBrowsePythonLib</sender>
   <signal>clicked()</signal>
   <receiver>DialogPreferences</receiver>
   <slot>browsePythonLib()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>608</x>
     <y>384</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdBrowseRepo</sender>
   <signal>clicked()</signal>
   <receiver>DialogPreferences</receiver>
   <slot>browseDeviserRepo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>605</x>
     <y>66</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdBrowseSpec</sender>
   <signal>clicked()</signal>
   <receiver>DialogPreferences</receiver>
   <slot>browseSbmlPkgSpec()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>605</x>
     <y>141</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdBrowseSwig</sender>
   <signal>clicked()</signal>
   <receiver>DialogPreferences</receiver>
   <slot>browseSWIG()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>608</x>
     <y>328</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdDelAttr</sender>
   <signal>clicked()</signal>
   <receiver>DialogPreferences</receiver>
   <slot>delType()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>42</x>
     <y>160</y>
    </hint>
    <hint type="destinationlabel">
     <x>207</x>
     <y>242</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdAddAttr</sender>
   <signal>clicked()</signal>
   <receiver>DialogPreferences</receiver>
   <slot>addType()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>42</x>
     <y>132</y>
    </hint>
    <hint type="destinationlabel">
     <x>207</x>
     <y>242</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>browsePython()</slot>
  <slot>browseDeviserRepo()</slot>
  <slot>browseOutputDir()</slot>
  <slot>browseSbmlPkgSpec()</slot>
  <slot>browseMikTex()</slot>
  <slot>browseLibSBML()</slot>
  <slot>selectCompiler()</slot>
  <slot>browseCMakeExecutable()</slot>
  <slot>browseDependencies()</slot>
  <slot>browseSWIG()</slot>
  <slot>browsePythonInclude()</slot>
  <slot>browsePythonLib()</slot>
  <slot>addType()</slot>
  <slot>delType()</slot>
 </slots>
</ui>
